### ISSUE TITLE
RFC: Dealing with -tag and -dev conda channels

### ISSUE THREAD

**Issue by ericdill** [NSLS-II/Bug-Reports#97]
I am running into a conda problem where metadatastore-0.3.2 is being preferentially picked up from the anaconda.org/lightsource2 or anaconda.nsls2.bnl.gov/nsls2-tag channel over the metadatastore-dev build which picks up the tag v0.3.0 and results in build names like `metadatastore-0.3.0.post67-67_g0d7129c_py27`.  This occurs even when i have the nsls2-dev or lightsource2-dev channels in a higher precedence than the nsls2-tag or lightsource2 channel.  

e.g., .condarc looks like this:

```
always_yes: true
binstar_upload: false
channels:
- ericdill
- lightsource2-dev
- lightsource2
- conda-forge
- defaults
show_channel_urls: true

track_features:
  - nomkl
```

This is really a problem when I try and conda-build the dev package for databroker which fails because it is picking up `metadatastore-0.3.2` from the tag channel instead of `metadatastore-v0.3.0.post.....` from the dev channel and, because there has been an API change, the conda package is failing to build.  I suppose this problem might be solved with a change to conda's package resolution procedure, currently under discussion: https://github.com/conda/conda/pull/2323

I suppose we could also solve this problem socially with a different tagging policy?  I guess I am advocating for something more like as soon as we spin off a maintenance branch (0.3.x) we immediately tag 0.4.0 or something, so that the dev builds from the master branch will always have a higher precedence than the maintenance builds 0.3.x.post-whatever or 0.3.2.

The final option would be to never have nsls2-tag and nsls2-dev in the same .condarc or we will get weird collisions like this.  I don't like this solution because there are a bunch of useful packages that we need to install the stack (like ffmpeg and friends) that are only available in nsls2-tag and not in nsls2-dev.

Basically what I want is a technical solution like what is being discussed in conda/conda#2323.  Until that lands, I think we might need to refine our tagging policy because I am currently unable to build the dev build of databroker internally :cry: 

Can I ask for some comments from the peanut gallery on this?  attn @klauer @danielballan @tacaswell @dchabot

**Comment by danielballan** [comment]
I propose removing 0.3.2 from the -tag channel until we are ready to deploy it.

**Comment by tacaswell** [comment]
The other thing that is weird here is that we are not merging the maintenance branch back into master.  The 'simple' fix here is to do a 'always ours' merge strategy merge from the maintenance branch into master.

**Comment by ericdill** [comment]
I think I like that plan @tacaswell , but could you clarify that statement a little?  I can't quite parse it.

**Comment by tacaswell** [comment]
I have done that merge and pushed to master.

The next dev build should see

```
19:06 $ git describe 
v0.3.2-62-g41491c3
```

which I think will fix your problem.  

The merge commit is https://github.com/NSLS-II/metadatastore/commit/41491c33ebaa05f162b4943c9538d02077d55076

**Comment by ericdill** [comment]
Awesome, thanks.  So what is the policy here?  Merge the maintenance branches back into master whenever there are new commits?

**Comment by ericdill** [comment]
Oh, or maybe it is just merge the maintenance branch back in when there are new tags?

**Comment by ericdill** [comment]
@tacaswell This is also a problem with databroker. Should I do the same thing and merge v0.3.x back into master?
